### ISSUE TITLE
OPRF IPA results sometimes don't exactly match expected results

### ISSUE THREAD

**Issue by benjaminsavage** [private-attribution/ipa#863]
This appears to be a new problem, introduced by https://github.com/private-attribution/ipa/pull/853

It's very hard to reproduce. I wasn't able to get it with 100, 1000, or 10000 rows of data. It wasn't until I ran with 100K rows of data that a few results were slightly different.

```
Running `target/debug/report_collector --network test_data/network.toml --input-file /tmp/events-100000.txt --output-file output.txt --disable-https oprf-ipa --max-breakdown-key 256 --per-user-credit-cap 32 --plaintext-match-keys`
2023-11-27T06:27:31.268358Z  INFO ipa_core::cli::verbosity: Logging setup at level info
2023-11-27T06:27:36.291803Z  INFO ipa_core::cli::playbook::ipa: Starting query for OPRF
2023-11-27T06:31:59.380573Z  INFO ipa_core::cli::playbook::ipa: Running IPA for 100000 records took 263.080834942s
2023-11-27T06:31:59.381055Z  WARN report_collector: output.txt file exists, renaming to "output-OjzH1"
2023-11-27T06:31:59.381681Z  INFO report_collector: IpaQueryConfig { per_user_credit_cap: 32, max_breakdown_key: 256, attribution_window_seconds: None, num_multi_bits: 3, plaintext_match_keys: true }
2023-11-27T06:31:59.382546Z  INFO ipa_core::cli::playbook: 
+-----+-----------+-----------+-------+
| Row | Expected  | Actual    | Diff? |
+=====================================+
| 0   | Some(498) | Some(498) |       |
|-----+-----------+-----------+-------|
| 1   | Some(371) | Some(371) |       |
|-----+-----------+-----------+-------|
| 2   | Some(376) | Some(376) |       |
|-----+-----------+-----------+-------|
| 3   | Some(433) | Some(433) |       |
|-----+-----------+-----------+-------|
| 4   | Some(529) | Some(529) |       |
|-----+-----------+-----------+-------|
| 5   | Some(422) | Some(422) |       |
|-----+-----------+-----------+-------|
| 6   | Some(407) | Some(407) |       |
|-----+-----------+-----------+-------|
| 7   | Some(507) | Some(507) |       |
|-----+-----------+-----------+-------|
| 8   | Some(402) | Some(402) |       |
|-----+-----------+-----------+-------|
| 9   | Some(394) | Some(394) |       |
|-----+-----------+-----------+-------|
| 10  | Some(392) | Some(392) |       |
|-----+-----------+-----------+-------|
| 11  | Some(451) | Some(451) |       |
|-----+-----------+-----------+-------|
| 12  | Some(378) | Some(378) |       |
|-----+-----------+-----------+-------|
| 13  | Some(444) | Some(444) |       |
|-----+-----------+-----------+-------|
| 14  | Some(377) | Some(377) |       |
|-----+-----------+-----------+-------|
| 15  | Some(449) | Some(449) |       |
|-----+-----------+-----------+-------|
| 16  | Some(455) | Some(455) |       |
|-----+-----------+-----------+-------|
| 17  | Some(543) | Some(543) |       |
|-----+-----------+-----------+-------|
| 18  | Some(501) | Some(501) |       |
|-----+-----------+-----------+-------|
| 19  | Some(492) | Some(492) |       |
|-----+-----------+-----------+-------|
| 20  | Some(462) | Some(462) |       |
|-----+-----------+-----------+-------|
| 21  | Some(482) | Some(482) |       |
|-----+-----------+-----------+-------|
| 22  | Some(490) | Some(490) |       |
|-----+-----------+-----------+-------|
| 23  | Some(486) | Some(488) | X     |
|-----+-----------+-----------+-------|
| 24  | Some(485) | Some(485) |       |
|-----+-----------+-----------+-------|
| 25  | Some(476) | Some(476) |       |
|-----+-----------+-----------+-------|
| 26  | Some(461) | Some(461) |       |
|-----+-----------+-----------+-------|
| 27  | Some(395) | Some(395) |       |
|-----+-----------+-----------+-------|
| 28  | Some(411) | Some(411) |       |
|-----+-----------+-----------+-------|
| 29  | Some(621) | Some(621) |       |
|-----+-----------+-----------+-------|
| 30  | Some(460) | Some(460) |       |
|-----+-----------+-----------+-------|
| 31  | Some(448) | Some(448) |       |
|-----+-----------+-----------+-------|
| 32  | Some(370) | Some(370) |       |
|-----+-----------+-----------+-------|
| 33  | Some(528) | Some(528) |       |
|-----+-----------+-----------+-------|
| 34  | Some(608) | Some(608) |       |
|-----+-----------+-----------+-------|
| 35  | Some(563) | Some(563) |       |
|-----+-----------+-----------+-------|
| 36  | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 37  | Some(449) | Some(449) |       |
|-----+-----------+-----------+-------|
| 38  | Some(508) | Some(508) |       |
|-----+-----------+-----------+-------|
| 39  | Some(496) | Some(496) |       |
|-----+-----------+-----------+-------|
| 40  | Some(495) | Some(495) |       |
|-----+-----------+-----------+-------|
| 41  | Some(448) | Some(448) |       |
|-----+-----------+-----------+-------|
| 42  | Some(498) | Some(498) |       |
|-----+-----------+-----------+-------|
| 43  | Some(472) | Some(472) |       |
|-----+-----------+-----------+-------|
| 44  | Some(471) | Some(471) |       |
|-----+-----------+-----------+-------|
| 45  | Some(502) | Some(502) |       |
|-----+-----------+-----------+-------|
| 46  | Some(435) | Some(435) |       |
|-----+-----------+-----------+-------|
| 47  | Some(420) | Some(420) |       |
|-----+-----------+-----------+-------|
| 48  | Some(429) | Some(429) |       |
|-----+-----------+-----------+-------|
| 49  | Some(550) | Some(550) |       |
|-----+-----------+-----------+-------|
| 50  | Some(494) | Some(494) |       |
|-----+-----------+-----------+-------|
| 51  | Some(516) | Some(516) |       |
|-----+-----------+-----------+-------|
| 52  | Some(559) | Some(559) |       |
|-----+-----------+-----------+-------|
| 53  | Some(589) | Some(589) |       |
|-----+-----------+-----------+-------|
| 54  | Some(392) | Some(392) |       |
|-----+-----------+-----------+-------|
| 55  | Some(390) | Some(390) |       |
|-----+-----------+-----------+-------|
| 56  | Some(464) | Some(467) | X     |
|-----+-----------+-----------+-------|
| 57  | Some(448) | Some(448) |       |
|-----+-----------+-----------+-------|
| 58  | Some(406) | Some(406) |       |
|-----+-----------+-----------+-------|
| 59  | Some(429) | Some(429) |       |
|-----+-----------+-----------+-------|
| 60  | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 61  | Some(450) | Some(450) |       |
|-----+-----------+-----------+-------|
| 62  | Some(460) | Some(460) |       |
|-----+-----------+-----------+-------|
| 63  | Some(414) | Some(414) |       |
|-----+-----------+-----------+-------|
| 64  | Some(507) | Some(507) |       |
|-----+-----------+-----------+-------|
| 65  | Some(479) | Some(479) |       |
|-----+-----------+-----------+-------|
| 66  | Some(458) | Some(458) |       |
|-----+-----------+-----------+-------|
| 67  | Some(526) | Some(526) |       |
|-----+-----------+-----------+-------|
| 68  | Some(520) | Some(520) |       |
|-----+-----------+-----------+-------|
| 69  | Some(407) | Some(407) |       |
|-----+-----------+-----------+-------|
| 70  | Some(511) | Some(511) |       |
|-----+-----------+-----------+-------|
| 71  | Some(484) | Some(484) |       |
|-----+-----------+-----------+-------|
| 72  | Some(347) | Some(347) |       |
|-----+-----------+-----------+-------|
| 73  | Some(323) | Some(323) |       |
|-----+-----------+-----------+-------|
| 74  | Some(390) | Some(390) |       |
|-----+-----------+-----------+-------|
| 75  | Some(579) | Some(579) |       |
|-----+-----------+-----------+-------|
| 76  | Some(420) | Some(420) |       |
|-----+-----------+-----------+-------|
| 77  | Some(352) | Some(352) |       |
|-----+-----------+-----------+-------|
| 78  | Some(478) | Some(478) |       |
|-----+-----------+-----------+-------|
| 79  | Some(581) | Some(581) |       |
|-----+-----------+-----------+-------|
| 80  | Some(427) | Some(427) |       |
|-----+-----------+-----------+-------|
| 81  | Some(490) | Some(490) |       |
|-----+-----------+-----------+-------|
| 82  | Some(444) | Some(444) |       |
|-----+-----------+-----------+-------|
| 83  | Some(465) | Some(465) |       |
|-----+-----------+-----------+-------|
| 84  | Some(577) | Some(577) |       |
|-----+-----------+-----------+-------|
| 85  | Some(440) | Some(440) |       |
|-----+-----------+-----------+-------|
| 86  | Some(390) | Some(390) |       |
|-----+-----------+-----------+-------|
| 87  | Some(378) | Some(378) |       |
|-----+-----------+-----------+-------|
| 88  | Some(401) | Some(401) |       |
|-----+-----------+-----------+-------|
| 89  | Some(341) | Some(341) |       |
|-----+-----------+-----------+-------|
| 90  | Some(492) | Some(492) |       |
|-----+-----------+-----------+-------|
| 91  | Some(483) | Some(483) |       |
|-----+-----------+-----------+-------|
| 92  | Some(453) | Some(453) |       |
|-----+-----------+-----------+-------|
| 93  | Some(508) | Some(508) |       |
|-----+-----------+-----------+-------|
| 94  | Some(381) | Some(381) |       |
|-----+-----------+-----------+-------|
| 95  | Some(402) | Some(402) |       |
|-----+-----------+-----------+-------|
| 96  | Some(495) | Some(495) |       |
|-----+-----------+-----------+-------|
| 97  | Some(501) | Some(501) |       |
|-----+-----------+-----------+-------|
| 98  | Some(472) | Some(472) |       |
|-----+-----------+-----------+-------|
| 99  | Some(447) | Some(447) |       |
|-----+-----------+-----------+-------|
| 100 | Some(576) | Some(576) |       |
|-----+-----------+-----------+-------|
| 101 | Some(416) | Some(416) |       |
|-----+-----------+-----------+-------|
| 102 | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 103 | Some(381) | Some(381) |       |
|-----+-----------+-----------+-------|
| 104 | Some(476) | Some(476) |       |
|-----+-----------+-----------+-------|
| 105 | Some(470) | Some(470) |       |
|-----+-----------+-----------+-------|
| 106 | Some(423) | Some(423) |       |
|-----+-----------+-----------+-------|
| 107 | Some(427) | Some(427) |       |
|-----+-----------+-----------+-------|
| 108 | Some(432) | Some(432) |       |
|-----+-----------+-----------+-------|
| 109 | Some(587) | Some(587) |       |
|-----+-----------+-----------+-------|
| 110 | Some(458) | Some(458) |       |
|-----+-----------+-----------+-------|
| 111 | Some(501) | Some(501) |       |
|-----+-----------+-----------+-------|
| 112 | Some(696) | Some(696) |       |
|-----+-----------+-----------+-------|
| 113 | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 114 | Some(512) | Some(512) |       |
|-----+-----------+-----------+-------|
| 115 | Some(453) | Some(453) |       |
|-----+-----------+-----------+-------|
| 116 | Some(632) | Some(632) |       |
|-----+-----------+-----------+-------|
| 117 | Some(581) | Some(581) |       |
|-----+-----------+-----------+-------|
| 118 | Some(504) | Some(504) |       |
|-----+-----------+-----------+-------|
| 119 | Some(545) | Some(545) |       |
|-----+-----------+-----------+-------|
| 120 | Some(499) | Some(499) |       |
|-----+-----------+-----------+-------|
| 121 | Some(377) | Some(377) |       |
|-----+-----------+-----------+-------|
| 122 | Some(425) | Some(425) |       |
|-----+-----------+-----------+-------|
| 123 | Some(683) | Some(683) |       |
|-----+-----------+-----------+-------|
| 124 | Some(440) | Some(440) |       |
|-----+-----------+-----------+-------|
| 125 | Some(555) | Some(555) |       |
|-----+-----------+-----------+-------|
| 126 | Some(325) | Some(325) |       |
|-----+-----------+-----------+-------|
| 127 | Some(328) | Some(328) |       |
|-----+-----------+-----------+-------|
| 128 | Some(485) | Some(485) |       |
|-----+-----------+-----------+-------|
| 129 | Some(523) | Some(523) |       |
|-----+-----------+-----------+-------|
| 130 | Some(501) | Some(501) |       |
|-----+-----------+-----------+-------|
| 131 | Some(468) | Some(468) |       |
|-----+-----------+-----------+-------|
| 132 | Some(479) | Some(479) |       |
|-----+-----------+-----------+-------|
| 133 | Some(451) | Some(451) |       |
|-----+-----------+-----------+-------|
| 134 | Some(606) | Some(606) |       |
|-----+-----------+-----------+-------|
| 135 | Some(456) | Some(456) |       |
|-----+-----------+-----------+-------|
| 136 | Some(398) | Some(398) |       |
|-----+-----------+-----------+-------|
| 137 | Some(494) | Some(494) |       |
|-----+-----------+-----------+-------|
| 138 | Some(448) | Some(448) |       |
|-----+-----------+-----------+-------|
| 139 | Some(484) | Some(484) |       |
|-----+-----------+-----------+-------|
| 140 | Some(414) | Some(414) |       |
|-----+-----------+-----------+-------|
| 141 | Some(452) | Some(452) |       |
|-----+-----------+-----------+-------|
| 142 | Some(483) | Some(483) |       |
|-----+-----------+-----------+-------|
| 143 | Some(437) | Some(437) |       |
|-----+-----------+-----------+-------|
| 144 | Some(479) | Some(479) |       |
|-----+-----------+-----------+-------|
| 145 | Some(401) | Some(401) |       |
|-----+-----------+-----------+-------|
| 146 | Some(556) | Some(556) |       |
|-----+-----------+-----------+-------|
| 147 | Some(495) | Some(495) |       |
|-----+-----------+-----------+-------|
| 148 | Some(412) | Some(412) |       |
|-----+-----------+-----------+-------|
| 149 | Some(611) | Some(611) |       |
|-----+-----------+-----------+-------|
| 150 | Some(458) | Some(458) |       |
|-----+-----------+-----------+-------|
| 151 | Some(516) | Some(516) |       |
|-----+-----------+-----------+-------|
| 152 | Some(467) | Some(467) |       |
|-----+-----------+-----------+-------|
| 153 | Some(346) | Some(346) |       |
|-----+-----------+-----------+-------|
| 154 | Some(522) | Some(522) |       |
|-----+-----------+-----------+-------|
| 155 | Some(499) | Some(499) |       |
|-----+-----------+-----------+-------|
| 156 | Some(382) | Some(382) |       |
|-----+-----------+-----------+-------|
| 157 | Some(510) | Some(510) |       |
|-----+-----------+-----------+-------|
| 158 | Some(503) | Some(503) |       |
|-----+-----------+-----------+-------|
| 159 | Some(438) | Some(438) |       |
|-----+-----------+-----------+-------|
| 160 | Some(574) | Some(574) |       |
|-----+-----------+-----------+-------|
| 161 | Some(450) | Some(450) |       |
|-----+-----------+-----------+-------|
| 162 | Some(458) | Some(458) |       |
|-----+-----------+-----------+-------|
| 163 | Some(421) | Some(421) |       |
|-----+-----------+-----------+-------|
| 164 | Some(419) | Some(419) |       |
|-----+-----------+-----------+-------|
| 165 | Some(518) | Some(518) |       |
|-----+-----------+-----------+-------|
| 166 | Some(486) | Some(486) |       |
|-----+-----------+-----------+-------|
| 167 | Some(483) | Some(483) |       |
|-----+-----------+-----------+-------|
| 168 | Some(453) | Some(453) |       |
|-----+-----------+-----------+-------|
| 169 | Some(464) | Some(464) |       |
|-----+-----------+-----------+-------|
| 170 | Some(408) | Some(408) |       |
|-----+-----------+-----------+-------|
| 171 | Some(429) | Some(429) |       |
|-----+-----------+-----------+-------|
| 172 | Some(454) | Some(454) |       |
|-----+-----------+-----------+-------|
| 173 | Some(422) | Some(422) |       |
|-----+-----------+-----------+-------|
| 174 | Some(454) | Some(454) |       |
|-----+-----------+-----------+-------|
| 175 | Some(451) | Some(451) |       |
|-----+-----------+-----------+-------|
| 176 | Some(500) | Some(500) |       |
|-----+-----------+-----------+-------|
| 177 | Some(525) | Some(525) |       |
|-----+-----------+-----------+-------|
| 178 | Some(397) | Some(397) |       |
|-----+-----------+-----------+-------|
| 179 | Some(471) | Some(471) |       |
|-----+-----------+-----------+-------|
| 180 | Some(444) | Some(444) |       |
|-----+-----------+-----------+-------|
| 181 | Some(384) | Some(384) |       |
|-----+-----------+-----------+-------|
| 182 | Some(415) | Some(415) |       |
|-----+-----------+-----------+-------|
| 183 | Some(294) | Some(294) |       |
|-----+-----------+-----------+-------|
| 184 | Some(402) | Some(402) |       |
|-----+-----------+-----------+-------|
| 185 | Some(556) | Some(556) |       |
|-----+-----------+-----------+-------|
| 186 | Some(503) | Some(503) |       |
|-----+-----------+-----------+-------|
| 187 | Some(424) | Some(424) |       |
|-----+-----------+-----------+-------|
| 188 | Some(472) | Some(472) |       |
|-----+-----------+-----------+-------|
| 189 | Some(400) | Some(400) |       |
|-----+-----------+-----------+-------|
| 190 | Some(493) | Some(493) |       |
|-----+-----------+-----------+-------|
| 191 | Some(436) | Some(436) |       |
|-----+-----------+-----------+-------|
| 192 | Some(583) | Some(583) |       |
|-----+-----------+-----------+-------|
| 193 | Some(490) | Some(490) |       |
|-----+-----------+-----------+-------|
| 194 | Some(389) | Some(389) |       |
|-----+-----------+-----------+-------|
| 195 | Some(486) | Some(486) |       |
|-----+-----------+-----------+-------|
| 196 | Some(354) | Some(354) |       |
|-----+-----------+-----------+-------|
| 197 | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 198 | Some(361) | Some(361) |       |
|-----+-----------+-----------+-------|
| 199 | Some(382) | Some(382) |       |
|-----+-----------+-----------+-------|
| 200 | Some(578) | Some(578) |       |
|-----+-----------+-----------+-------|
| 201 | Some(350) | Some(350) |       |
|-----+-----------+-----------+-------|
| 202 | Some(434) | Some(434) |       |
|-----+-----------+-----------+-------|
| 203 | Some(445) | Some(445) |       |
|-----+-----------+-----------+-------|
| 204 | Some(406) | Some(406) |       |
|-----+-----------+-----------+-------|
| 205 | Some(446) | Some(446) |       |
|-----+-----------+-----------+-------|
| 206 | Some(492) | Some(492) |       |
|-----+-----------+-----------+-------|
| 207 | Some(373) | Some(373) |       |
|-----+-----------+-----------+-------|
| 208 | Some(426) | Some(426) |       |
|-----+-----------+-----------+-------|
| 209 | Some(557) | Some(557) |       |
|-----+-----------+-----------+-------|
| 210 | Some(450) | Some(450) |       |
|-----+-----------+-----------+-------|
| 211 | Some(445) | Some(445) |       |
|-----+-----------+-----------+-------|
| 212 | Some(472) | Some(472) |       |
|-----+-----------+-----------+-------|
| 213 | Some(436) | Some(436) |       |
|-----+-----------+-----------+-------|
| 214 | Some(490) | Some(490) |       |
|-----+-----------+-----------+-------|
| 215 | Some(493) | Some(493) |       |
|-----+-----------+-----------+-------|
| 216 | Some(447) | Some(447) |       |
|-----+-----------+-----------+-------|
| 217 | Some(524) | Some(524) |       |
|-----+-----------+-----------+-------|
| 218 | Some(463) | Some(469) | X     |
|-----+-----------+-----------+-------|
| 219 | Some(464) | Some(464) |       |
|-----+-----------+-----------+-------|
| 220 | Some(462) | Some(462) |       |
|-----+-----------+-----------+-------|
| 221 | Some(479) | Some(479) |       |
|-----+-----------+-----------+-------|
| 222 | Some(429) | Some(429) |       |
|-----+-----------+-----------+-------|
| 223 | Some(422) | Some(422) |       |
|-----+-----------+-----------+-------|
| 224 | Some(484) | Some(484) |       |
|-----+-----------+-----------+-------|
| 225 | Some(479) | Some(479) |       |
|-----+-----------+-----------+-------|
| 226 | Some(391) | Some(391) |       |
|-----+-----------+-----------+-------|
| 227 | Some(560) | Some(560) |       |
|-----+-----------+-----------+-------|
| 228 | Some(448) | Some(448) |       |
|-----+-----------+-----------+-------|
| 229 | Some(463) | Some(463) |       |
|-----+-----------+-----------+-------|
| 230 | Some(352) | Some(352) |       |
|-----+-----------+-----------+-------|
| 231 | Some(483) | Some(483) |       |
|-----+-----------+-----------+-------|
| 232 | Some(391) | Some(391) |       |
|-----+-----------+-----------+-------|
| 233 | Some(451) | Some(451) |       |
|-----+-----------+-----------+-------|
| 234 | Some(600) | Some(600) |       |
|-----+-----------+-----------+-------|
| 235 | Some(374) | Some(374) |       |
|-----+-----------+-----------+-------|
| 236 | Some(413) | Some(413) |       |
|-----+-----------+-----------+-------|
| 237 | Some(400) | Some(400) |       |
|-----+-----------+-----------+-------|
| 238 | Some(444) | Some(444) |       |
|-----+-----------+-----------+-------|
| 239 | Some(485) | Some(485) |       |
|-----+-----------+-----------+-------|
| 240 | Some(473) | Some(473) |       |
|-----+-----------+-----------+-------|
| 241 | Some(473) | Some(473) |       |
|-----+-----------+-----------+-------|
| 242 | Some(309) | Some(309) |       |
|-----+-----------+-----------+-------|
| 243 | Some(373) | Some(373) |       |
|-----+-----------+-----------+-------|
| 244 | Some(405) | Some(405) |       |
|-----+-----------+-----------+-------|
| 245 | Some(457) | Some(457) |       |
|-----+-----------+-----------+-------|
| 246 | Some(314) | Some(314) |       |
|-----+-----------+-----------+-------|
| 247 | Some(378) | Some(378) |       |
|-----+-----------+-----------+-------|
| 248 | Some(453) | Some(453) |       |
|-----+-----------+-----------+-------|
| 249 | Some(402) | Some(402) |       |
|-----+-----------+-----------+-------|
| 250 | Some(455) | Some(455) |       |
|-----+-----------+-----------+-------|
| 251 | Some(462) | Some(462) |       |
|-----+-----------+-----------+-------|
| 252 | Some(546) | Some(546) |       |
|-----+-----------+-----------+-------|
| 253 | Some(573) | Some(573) |       |
|-----+-----------+-----------+-------|
| 254 | Some(439) | Some(439) |       |
|-----+-----------+-----------+-------|
| 255 | Some(454) | Some(454) |       |
+-----+-----------+-----------+-------+

2023-11-27T06:31:59.390625Z ERROR ipa_core::error: ThreadId(1) "main" panicked at /Users/btsavage/projects/raw-ipa/ipa-core/src/cli/playbook/mod.rs:60:5:
Expected and actual results don't match: [(23, Some(486), Some(488)), (56, Some(464), Some(467)), (218, Some(463), Some(469))]
```

**Comment by benjaminsavage** [comment]
I was able to reduce this down to the following minimal reproducible test case:
```
29135,10251308645,0,218,0
179334,10251308645,1,0,3
313530,10251308645,1,0,3
327956,10251308645,1,0,5
368968,10251308645,1,0,6
453715,10251308645,1,0,1
471555,10251308645,1,0,2
530458,10251308645,1,0,6
536295,10251308645,1,0,6
600360,10251308645,1,0,6
```

**Comment by benjaminsavage** [comment]
Fixed by #864
